### PR TITLE
Adding import to resources

### PR DESCRIPTION
**Pull request by byteford**
Added import ability and tests for all implemented resources

### CODE REVIEW

File: aquasec/resource_application_scope.go [hunk 1]
```diff
@@ -402,14 +405,14 @@ func expandApplicationScope(d *schema.ResourceData) (*client.ApplicationScope, e
...
-
-	iap, err := ac.GetApplicationScope(name)
+	iap, err := ac.GetApplicationScope(d.Id())
 	if err == nil {
+
 		d.Set("name", iap.Name)
 		d.Set("description", iap.Description)
 		d.Set("author", iap.Author)
 		d.Set("owner_email", iap.OwnerEmail)
+		d.SetId(iap.Name)
```
Comment (yossig-aquasec) [inline aquasec/resource_application_scope.go:415]: why to set it twice? first time is in line: 341

File: aquasec/resource_application_scope.go [hunk 2]
```diff
@@ -335,10 +338,10 @@ func resourceApplicationScopeCreate(d *schema.ResourceData, m interface{}) error
 
 	if err == nil {
 		d.Set("categories", flattenCategories(iap.Categories))
-
+		d.SetId(name)
 		err1 := resourceApplicationScopeRead(d, m)
 		if err1 == nil {
-			d.SetId(name)
+
```
Comment (yossig-aquasec) [inline aquasec/resource_application_scope.go:344]: please change it to if err1 != true {return fmt.Errorf("application scope resource read is failed with error: %v", err1)}

File: aquasec/resource_group.go [hunk 3]
```diff
@@ -52,23 +54,21 @@ func resourceGroupCreate(d *schema.ResourceData, m interface{}) error {
...
 
-	id := d.Get("group_id").(int)
+	id, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return nil
+	}
 	r, err := ac.GetGroup(id)
 	if err == nil {
+		d.Set("name", r.Name)
+		d.Set("group_id", r.Id)
```
Comment (yossig-aquasec) [inline aquasec/resource_group.go:71]: duplicate set for group_id and id

File: aquasec/resource_host_assurance_policy.go [hunk 4]
```diff
@@ -660,7 +656,7 @@ func resourceHostAssurancePolicyUpdate(d *schema.ResourceData, m interface{}) er
 		if err == nil {
 			err1 := resourceHostAssurancePolicyRead(d, m)
 			if err1 == nil {
-				d.SetId(name)
+				d.SetId(iap.Name)
```
Comment (yossig-aquasec) [inline aquasec/resource_host_assurance_policy.go:659]: duplicate it set, also in line 638

File: aquasec/resource_host_runtime_policy.go [hunk 5]
```diff
@@ -496,7 +494,7 @@ func resourceHostRuntimePolicyRead(ctx context.Context, d *schema.ResourceData,
 		d.Set("windows_registry_monitoring", flattenWindowsRegistryMonitoring(crp.RegistryAccessMonitoring))
 		d.Set("windows_registry_protection", flattenWindowsRegistryProtection(crp.ReadonlyRegistry))
```
Comment (yossig-aquasec) [inline aquasec/resource_host_runtime_policy.go:496]: duplicate set id, also in line: 460

File: aquasec/resource_image.go [hunk 6]
```diff
@@ -698,30 +702,24 @@ func resourceImageCreate(ctx context.Context, d *schema.ResourceData, m interfac
...
 
-	return nil
-	//return resourceImageRead(ctx, d, m)
 }
 
 func resourceImageRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var err error
 	c := m.(*client.Client)
-	image := expandImage(d)
+	id := d.Id()
```
Comment (yossig-aquasec) [inline aquasec/resource_image.go:713]: duplicate set id, also in line: 705

File: aquasec/resource_user_saas.go [hunk 7]
```diff
@@ -196,50 +197,44 @@ func resourceUserSaasCreate(d *schema.ResourceData, m interface{}) error {
...
+		l := make(map[string]interface{})
+		l["id"] = login.Id
+		l["ip_address"] = login.IpAddress
+		l["created"] = login.Created
+		l["user_id"] = login.UserId
+		logins[i] = l
+	}
+
+	d.Set("logins", logins)
+	d.SetId(r.BasicId.Id)
```
Comment (yossig-aquasec) [inline aquasec/resource_user_saas.go:237]: duplicate set id, also in line: 200

**Review by yossig-aquasec — CHANGES_REQUESTED** [review]
@byteford  please look at my comments

File: aquasec/resource_user_saas.go (hunk 7 above)
Comment (byteford) [reply aquasec/resource_user_saas.go:237]: like 200 is part of create, this is in the read function?

File: aquasec/resource_group.go (hunk 3 above)
Comment (byteford) [reply aquasec/resource_group.go:71]: what do you mean by duplicate set? group_id is part of the state and id is something terraform uses to track the resource

File: aquasec/resource_host_assurance_policy.go (hunk 4 above)
Comment (byteford) [reply aquasec/resource_host_assurance_policy.go:659]: 638 is part of the create function this is the update function?

File: aquasec/resource_host_runtime_policy.go (hunk 5 above)
Comment (byteford) [reply aquasec/resource_host_runtime_policy.go:496]: 460 is the create function, this is the read function?

File: aquasec/resource_image.go (hunk 6 above)
Comment (byteford) [reply aquasec/resource_image.go:713]: 705 is the create function this is the read function?

File: aquasec/resource_application_scope.go (hunk 2 above)
Comment (byteford) [reply aquasec/resource_application_scope.go:344]: have pushed a change to improve this code

File: aquasec/resource_application_scope.go (hunk 1 above)
Comment (byteford) [reply aquasec/resource_application_scope.go:415]: 341 is create function this is read?

Comment (yossig-aquasec) [reply aquasec/resource_application_scope.go:416]: yes, but the only use with the read is after create, so basically you are setting it twice

Comment (byteford) [reply aquasec/resource_application_scope.go:416]: the read is used with import as well as being called after read

**Review by yossig-aquasec — APPROVED** [review]

**Pull request merged**